### ISSUE TITLE
Bug: Calling Pandas `get_as_df()` on query result causes an error

### ISSUE THREAD

**Issue by marcosbodio** [kuzudb/kuzu#3992]
### Kùzu version

0.4.2

### What operating system are you using?

Mac OSX Sonoma 14.5 on Apple M1 Max chip

### What happened?

Using example at https://kuzudb.com/api-docs/python/kuzu.html, but calling `get_as_df()`:
```
import kuzu

db = kuzu.Database("./test")
conn = kuzu.Connection(db)

# Define the schema
conn.execute("CREATE NODE TABLE User(name STRING, age INT64, PRIMARY KEY (name))")
conn.execute("CREATE NODE TABLE City(name STRING, population INT64, PRIMARY KEY (name))")
conn.execute("CREATE REL TABLE Follows(FROM User TO User, since INT64)")
conn.execute("CREATE REL TABLE LivesIn(FROM User TO City)")

# Load some data
conn.execute('COPY User FROM "user.csv"')
conn.execute('COPY City FROM "city.csv"')
conn.execute('COPY Follows FROM "follows.csv"')
conn.execute('COPY LivesIn FROM "lives-in.csv"')

# Query the data
results = conn.execute("MATCH (u:User) RETURN u.name, u.age;")
df = results.get_as_df()
print(df)
```

I get the following error:
- when running from PyCharm: Process finished with exit code 139 (interrupted by signal 11:SIGSEGV)
- when running from command line: terminated by signal SIGSEGV (Address boundary error)

When executing the code, the database is created, but the call to `get_as_df()` fails.

Environment:
- Mac OSX Sonoma 14.5 on Apple M1 Max chip
- Python 3.11.9
- cat requirements.txt
kuzu==0.4.2
matplotlib==3.9.1
networkx==3.3
pandas==2.2.2

### Are there known steps to reproduce?

See above.

**Comment by prrao87** [comment]
Hi @marcosbodio could you rerun the same code with the following version of numpy installed?

pip install numpy~=1.26.0

**Comment by marcosbodio** [comment]
Hi @prrao87, i did run the same code with with the `numpy` version that you suggested and it works perfectly.

**Comment by prrao87** [comment]
Great! I'll leave this to @mxwli to mark this as closed if/when we figure out our next steps for version 0.5.0 (we do provide a much better error message in the upcoming release, so that should help users know to do this without having to read through the docs).

**Comment by mxwli** [comment]
The issue is fixed by #3839 
It'll be released with 0.5.0, but you can try the nightly build to test if it works. Let us know if there are any subsequent problems with numpy
